lib/SampleService/core/api_translation.py

```diff
@@ -612,3 +606,20 @@ def links_to_dicts(links: List[DataLink]) -> List[Dict[str, Any]]:
             'expired': ex
         })
     return ret
+
+def validate_sample_id(id_, name=None):
```

Comment:
Does it make sense for the `name` parameter to be optional? You'll just wind up with a nonsensical error message if it's left out.